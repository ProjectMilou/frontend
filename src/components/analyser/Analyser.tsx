import React from 'react';

const Analyser: React.FC = () => <div>Analyser component</div>;

export default Analyser;
