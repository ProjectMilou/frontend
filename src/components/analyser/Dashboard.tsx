import React from 'react';
import {
  LinearProgress,
  makeStyles,
  Container,
} from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import * as API from '../../portfolio/APIClient';
import { ErrorCode } from '../../Errors';
import ErrorMessage from './ErrorMessage';
import PortfolioOverview from './StockListOverview';
import DashboardHeader from './DashboardHeader';

export type DashboardProps = {
  token: string;
  selectPortfolio: (id: string) => void;
};

const useStyles = makeStyles({
  createButton: {
    marginTop: '25px',
  },
  dashboard: {
    margin: '25px auto',
  },
});

const Dashboard: React.FC<DashboardProps> = ({ token, selectPortfolio }) => {
  const [portfolios, setPortfolios] = React.useState<API.PortfolioOverview[]>();
  const [error, setError] = React.useState<ErrorCode | undefined>();

  const isMounted = React.useRef(true);
  const fetch = async () => {
    setError(undefined);
    try {
      const p = await API.portfolioOverview(token);
      if (isMounted.current) {
        setPortfolios(p);
      }
    } catch (e) {
      if (isMounted.current) {
        setError(e.message);
      }
    }
  };

  React.useEffect(() => {
    fetch();
    return () => {
      isMounted.current = false;
    };
    // deps must be empty because the function should only be called on mount.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const classes = useStyles();

  return (
    <>
      <DashboardHeader />
      {!portfolios && !error && (
        <div>
          <LinearProgress color="secondary" />
        </div>
      )}
      <Container maxWidth="lg" className={classes.dashboard}>
        {error && (
          <ErrorMessage
            error={error}
            messageKey="analyser.dashboard.errorMessage"
            handling={
              error.startsWith('AUTH')
                ? {
                    buttonText: 'error.action.login',
                    action: async () => {
                      // TODO: go back to login
                    },
                  }
                : {
                    buttonText: 'error.action.retry',
                    action: fetch,
                  }
            }
          />
        )}
        {portfolios && (
          <div>
            <PortfolioOverview
              portfolios={portfolios}
              selectPortfolio={selectPortfolio}
            />
          </div>
        )}
      </Container>
    </>
  );
};

export default Dashboard;
