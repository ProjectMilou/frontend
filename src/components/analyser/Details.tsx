import React from 'react';

// props type declaration
export type DetailsProps = {
  // function to return to the dashboard
  back: () => void;
};

const Details: React.FC<DetailsProps> = () => (
  <div>
    details will be here
  </div>
);

export default Details;
