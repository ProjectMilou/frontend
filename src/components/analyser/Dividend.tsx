import React from 'react'
import { createStyles, Theme, makeStyles } from '@material-ui/core/styles';
import DividendsChart from "../shared_components/DividendsChart";
import DonutChart from '../shared_components/DonutChart';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '100%',
      maxWidth: 360,
      backgroundColor: theme.palette.background.paper,
    },
    indented: {
        paddingLeft: theme.spacing(7),
    },
    rowC: {
      display: "flex",
      flexDirection: "row",
      justifyContent: 'space-evenly'
    }
  }),
);

const Dividend: React.FC = () => {
    const classes = useStyles();
    return (
        <div className={classes.indented}>
            <h1>Dividends</h1>
            <div className={classes.rowC}>
                <DividendsChart/>
                <DonutChart/>
            </div>
        </div>
    )
} 
    
    
export default Dividend