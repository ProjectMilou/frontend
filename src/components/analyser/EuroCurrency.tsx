import React from 'react';
import NumberFormat from 'react-number-format';
import { makeStyles, Theme } from '@material-ui/core';

export type EuroCurrencyProps = {
  value: number;
  size?: string;
  fontWeight?: number;
  color?: string;
};

const useStyles = makeStyles<Theme, EuroCurrencyProps, string>({
  euroCurrency: {
    fontSize: (props) => props.size || '24px',
    fontWeight: (props)=> props.fontWeight || 400,
    color: (props)=> props.color || "#122654",
  },
});

const EuroCurrency: React.FC<EuroCurrencyProps> = (props) => {
  const { value } = props;
  const classes = useStyles(props);

  return (
    <NumberFormat
      value={value}
      displayType="text"
      prefix="â‚¬"
      decimalScale={2}
      fixedDecimalScale
      decimalSeparator="."
      className={classes.euroCurrency}
    />
  );
};

export default EuroCurrency;
