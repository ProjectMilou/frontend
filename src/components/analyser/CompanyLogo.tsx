import React from 'react';
import { Stock } from '../../analyser/APIClient';
import logo from '../../assets/images/placeholder.png';

type LogoProps = {
  stockOverview: Stock;
  style: string;
};

/**
 * This component displays the stock logos and replaces it with the Milou logo incase it doesn't exist
 * @param style allows the logo to be reused in different location independently styled
 */
const CompanyLogo: React.FC<LogoProps> = ({ stockOverview, style }) => {
  if (stockOverview.picture.toString() !== '') {
    return (
      <img
        className={style}
        alt="Company Logo"
        src={stockOverview.picture.toString()}
      />
    );
  }
  return <img className={style} alt="Company Logo" src={logo} />;
};

export default CompanyLogo;
