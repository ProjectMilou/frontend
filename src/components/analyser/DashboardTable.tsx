import React from 'react';
import { useTranslation } from 'react-i18next';
import {
  lighten,
  ListItemText,
  makeStyles,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Theme,
  Typography,
} from '@material-ui/core';
import classNames from 'classnames';
import * as API from '../../analyser/APIClient';
import EuroCurrency from './EuroCurrency';
import Performance from './Performance';

const useStyles = makeStyles((theme: Theme) => ({
  action: { display: 'inline-block' },
  row: {
    cursor: 'pointer',
  },
  rowHover: {
    backgroundColor: lighten(theme.palette.primary.light, 0.85),
  },
  positionCount: {
    fontSize: '24px',
  },
  disabled: {
    cursor: 'not-allowed',
  },
}));

export type DashboardTableRowProps = {
  stock: API.Stock;
};

export const DashboardTableRow: React.FC<DashboardTableRowProps> = ({
  stock
}) => {
  const [hover, setHover] = React.useState<boolean>(false);

  const { t } = useTranslation();
  const classes = useStyles();

  const selectStock = (symbol: API.Stock['symbol']) => {
    console.log(symbol)
  }

  return (
    <TableRow
      onClick={() => {
        selectStock(stock.symbol);
      }}
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}
      className={classNames(classes.row, hover && classes.rowHover)}
    >
      <TableCell align="center">{stock.name}</TableCell>
      <TableCell align="center" component="th" scope="row">
        <ListItemText
          primary={<Typography color="primary">{stock.symbol}</Typography>}
          secondary={
           stock['1d'] > 0 ? t('portfolio.virtual') : t('portfolio.real')
          }
          secondaryTypographyProps={{
            color: stock['1d'] > 0 ? 'textSecondary' : 'secondary',
          }}
        />
      </TableCell>
      <TableCell align="center" className={classes.positionCount}>
        {stock.analystTargetPrice}
      </TableCell>
      <TableCell align="center">
        <EuroCurrency value={stock.valuation} />
      </TableCell>
      <TableCell align="center">
        <Performance value={stock['7d']} />
      </TableCell>
      <TableCell align="center">
        <Performance value={stock['30d']} />
      </TableCell>
    </TableRow>
  );
};

export type DashboardTableProps = {
  stocks: API.Stock[];
};

const DashboardTable: React.FC<DashboardTableProps> = ({
  stocks
}) => {
  const { t } = useTranslation();

  // TODO: Improve portfolio score visualization
  return (
    <TableContainer component={Paper}>
      <Table aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell align="center">{t('portfolio.score')}</TableCell>
            <TableCell align="center">{t('portfolio.name')}</TableCell>
            <TableCell align="center">
              {t('portfolio.positionsCount')}
            </TableCell>
            <TableCell align="center">{t('portfolio.value')}</TableCell>
            <TableCell align="center">{t('portfolio.7d')}</TableCell>
            <TableCell align="center">{t('portfolio.1y')}</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {stocks.map((s) => (
            <DashboardTableRow
              stock={s}
              key={s.symbol}
            />
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

export default DashboardTable;
