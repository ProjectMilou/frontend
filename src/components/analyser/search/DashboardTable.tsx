// Based on Portfolio's DashboardTable.tsx Will be later either replaced by Material-UI list or refactored

import React from 'react';
import { useTranslation } from 'react-i18next';
import {
  lighten,
  ListItemText,
  makeStyles,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Theme,
  Typography,
  createStyles,
} from '@material-ui/core';
import classNames from 'classnames';
import * as API from '../../../analyser/APIClient';
import EuroCurrency from '../../shared/EuroCurrency';
import Performance from '../../shared/Performance';
import Valuation from '../../shared/Valuation';

const useStyles = makeStyles(({ palette }: Theme) =>
  createStyles({
    action: { display: 'inline-block' },
    row: {
      cursor: 'pointer',
    },
    rowHover: {
      backgroundColor: lighten(palette.primary.light, 0.85),
    },
    defaultText: {
      fontSize: '24px',
      color: palette.primary.main,
    },
    disabled: {
      cursor: 'not-allowed',
    },
  })
);

export type DashboardTableRowProps = {
  stock: API.Stock;
  selectStock: (id: string) => void;
};

// Rounds and adds M=Million, B=Billion and K=Thousand --> American System!!!
function moneyFormat(val: number): string {
  let round = '';
  if (Math.abs(val) >= 1.0e9) {
    round = `€${Math.round(Math.abs(val) / 1.0e9)}B`;
  } else if (Math.abs(val) >= 1.0e6) {
    round = `€${Math.round(Math.abs(val) / 1.0e6)}M`;
  } else if (Math.abs(val) >= 1.0e3) {
    round = `€${Math.round(Math.abs(val) / 1.0e3)}K`;
  } else {
    round = `€${Math.abs(val)}`;
  }
  return round;
}

// Numbers are output in different colors need clear approach!
export const DashboardTableRow: React.FC<DashboardTableRowProps> = ({
  stock,
  selectStock,
}) => {
  const [hover, setHover] = React.useState<boolean>(false);

  const { t } = useTranslation();
  const classes = useStyles();

  return (
    <TableRow
      onClick={() => {
        selectStock(stock.symbol);
      }}
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}
      className={classNames(classes.row, hover && classes.rowHover)}
    >
      <TableCell align="center" component="th" scope="row">
        <ListItemText
          primary={
            <Typography className={classes.defaultText} color="primary">
              {stock.symbol}
            </Typography>
          }
          secondary={stock.name}
          secondaryTypographyProps={{
            color: 'textSecondary',
          }}
        />
      </TableCell>
      <TableCell align="center" className={classes.defaultText}>
        <EuroCurrency
          value={stock.price}
          decimalSeperator="."
          thousandSeperator=","
        />
      </TableCell>
      <TableCell align="center">
        <Performance
          value={stock.per7d}
          decimalSeperator="."
          thousandSeperator=","
        />
      </TableCell>
      <TableCell align="center">
        <Performance
          value={stock.per30d}
          decimalSeperator="."
          thousandSeperator=","
        />
      </TableCell>
      <TableCell align="center">
        <Typography color="primary" className={classes.defaultText}>
          {moneyFormat(stock.marketCapitalization)}
        </Typography>
      </TableCell>
      <TableCell align="center">
        <EuroCurrency
          value={stock.analystTargetPrice}
          decimalSeperator="."
          thousandSeperator=","
        />
      </TableCell>
      <TableCell align="center">
        <Valuation value={stock.valuation} />
      </TableCell>
      <TableCell align="center">
        <Performance
          value={stock.div}
          decimalSeperator="."
          thousandSeperator=","
        />
      </TableCell>
      <TableCell align="center">
        <Typography color="primary" className={classes.defaultText}>
          {t(`${stock.industry}`)}
        </Typography>
      </TableCell>
    </TableRow>
  );
};

export type DashboardTableProps = {
  stocks: API.Stock[];
  selectStock: (symbol: string) => void;
};

const DashboardTable: React.FC<DashboardTableProps> = ({
  stocks,
  selectStock,
}) => {
  const { t } = useTranslation();

  return (
    <TableContainer component={Paper}>
      <Table aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell align="center">{t('stock.name')}</TableCell>
            <TableCell align="center">{t('stock.lastPrice')}</TableCell>
            <TableCell align="center">{t('stock.7d')}</TableCell>
            <TableCell align="center">{t('stock.30d')}</TableCell>
            <TableCell align="center">{t('stock.marketCap')}</TableCell>
            <TableCell align="center">{t('stock.analystsTarget')}</TableCell>
            <TableCell align="center">{t('stock.valuation')}</TableCell>
            <TableCell align="center">{t('stock.div')}</TableCell>
            <TableCell align="center">{t('stock.industry')}</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {stocks.map((s) => (
            <DashboardTableRow
              stock={s}
              selectStock={selectStock}
              key={s.symbol}
            />
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

export default DashboardTable;
