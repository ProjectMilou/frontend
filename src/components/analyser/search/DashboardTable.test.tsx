import { render, fireEvent } from '@testing-library/react';
import * as React from 'react';
import { createMuiTheme, ThemeProvider } from '@material-ui/core';
import DashboardTable, { DashboardTableProps } from './DashboardTable';
import { MockOverview, MockOverviewTwo } from '../../../analyser/APIMocks';

const theme = createMuiTheme({
  typography: {
    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      '"Helvetica Neue"',
      'Arial',
      'sans-serif',
      '"Apple Color Emoji"',
      '"Segoe UI Emoji"',
      '"Segoe UI Symbol"',
    ].join(','),
  },
  palette: {
    background: {
      default: '#EEF1FB',
    },
    primary: {
      light: '#415176',
      main: '#122654',
      dark: '#0c1a3a',
      contrastText: '#EEF1FB',
    },
    secondary: {
      light: '#df4f9b',
      main: '#D72483',
      dark: '#96195b',
      contrastText: '#000',
    },
    success: {
      main: '#50E2A8',
    },
    warning: {
      main: '#FFC43B',
    },
    error: {
      main: '#D64745',
    },
    lightBlue: {
      main: '#3FBCF2',
    },
    teal: {
      main: '#34CFB2',
    },
  },
});

describe('DashboardTable', () => {
  const defaultProps: DashboardTableProps = {
    stocks: [MockOverview, MockOverviewTwo],
  };

  const renderComponent = (newProps?: Partial<DashboardTableProps>) => {
    const props = { ...defaultProps, ...newProps };
    return {
      ...render(
        <ThemeProvider theme={theme}>
          <DashboardTable {...props} />
        </ThemeProvider>
      ),
      props,
    };
  };

  test('show empty table if no stocks exist', () => {
    const { container, queryByText } = renderComponent({ stocks: [] });
    // empty table & no error message
    expect(queryByText('analyser.dashboard.errorMessage')).toBeNull();
    expect(container.querySelector('table')).toBeInTheDocument();
    expect(container.querySelector('ul')).toBeNull();
  });

  test('display stock information', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  test('select stock on click', () => {
    const { props, getByText, queryByText } = renderComponent();
    fireEvent.click(getByText(props.stocks[0].symbol));
    expect(queryByText(props.stocks[0].symbol)).toBeInTheDocument();
    fireEvent.click(getByText(props.stocks[1].symbol));
    expect(queryByText(props.stocks[1].symbol)).toBeInTheDocument();
  });
});
