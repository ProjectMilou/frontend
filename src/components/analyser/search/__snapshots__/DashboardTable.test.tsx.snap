// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTable display stock information 1`] = `
<div>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: auto; overflow: auto;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiTableContainer-root makeStyles-customTableContainer-7"
          id="scrollableTable"
        >
          <table
            aria-label="simple table"
            class="MuiTable-root makeStyles-table-5 MuiTable-stickyHeader"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  aria-sort="ascending"
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                    role="button"
                    tabindex="0"
                  >
                    stock.name
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    stock.lastPrice
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    stock.7d
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    stock.365d
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    stock.marketCap
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-paddingNone MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    stock.analystsTarget
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    stock.valuation
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    stock.div
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root makeStyles-customTableHead-8 MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
                  scope="col"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiTableSortLabel-root"
                    role="button"
                    tabindex="0"
                  >
                    stock.industry
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root makeStyles-row-10"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  <div
                    class="makeStyles-compContainer-17"
                  >
                    <div
                      class="makeStyles-pWrapper-18 makeStyles-pWrapper-33"
                    >
                      <p
                        class="makeStyles-top-15 makeStyles-top-31"
                      >
                        APPLE
                      </p>
                    </div>
                    <div
                      class="makeStyles-pWrapper-18 makeStyles-pWrapper-33"
                    >
                      <p
                        class="makeStyles-bottom-16 makeStyles-bottom-32"
                      >
                        International Business Machines Corporation
                      </p>
                    </div>
                  </div>
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-34"
                  >
                    138,16 €
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-35"
                  >
                    1,05 %
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-36"
                  >
                    1,11 %
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  <p
                    class="MuiTypography-root makeStyles-defaultText-12 MuiTypography-body1 MuiTypography-colorPrimary"
                  >
                    €
                    119 B
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-37"
                  >
                    139,18 €
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-euroCurrency-28 makeStyles-euroCurrency-38"
                  >
                    21.4x
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-39"
                  >
                    6,51 %
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  <p
                    class="MuiTypography-root makeStyles-highlightText-13 MuiTypography-body1 MuiTypography-colorPrimary"
                  >
                    Information Technology Services
                  </p>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root makeStyles-row-10"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  <div
                    class="makeStyles-compContainer-17"
                  >
                    <div
                      class="makeStyles-pWrapper-18 makeStyles-pWrapper-21"
                    >
                      <p
                        class="makeStyles-top-15 makeStyles-top-19"
                      >
                        IBM
                      </p>
                    </div>
                    <div
                      class="makeStyles-pWrapper-18 makeStyles-pWrapper-21"
                    >
                      <p
                        class="makeStyles-bottom-16 makeStyles-bottom-20"
                      >
                        International Business Machines Corporation
                      </p>
                    </div>
                  </div>
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-24"
                  >
                    138,16 €
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-25"
                  >
                    1,05 %
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-26"
                  >
                    1,11 %
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  <p
                    class="MuiTypography-root makeStyles-defaultText-12 MuiTypography-body1 MuiTypography-colorPrimary"
                  >
                    €
                    119 B
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-27"
                  >
                    139,18 €
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-euroCurrency-28 makeStyles-euroCurrency-29"
                  >
                    21.4x
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body makeStyles-defaultText-12 MuiTableCell-alignCenter"
                >
                  <span
                    class="makeStyles-styledNumberFormat-22 makeStyles-styledNumberFormat-30"
                  >
                    6,51 %
                  </span>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                >
                  <p
                    class="MuiTypography-root makeStyles-highlightText-13 MuiTypography-body1 MuiTypography-colorPrimary"
                  >
                    Information Technology Services
                  </p>
                </td>
              </tr>
              <p
                class="MuiTypography-root makeStyles-loading-6 MuiTypography-body1"
              >
                <div
                  class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                  role="progressbar"
                  style="width: 40px; height: 40px;"
                >
                  <svg
                    class="MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </div>
              </p>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
